app/src/main/java/com/fastaccess/ui/modules/search/SearchUserActivity.kt

```diff
@@ -0,0 +1,118 @@
+package com.fastaccess.ui.modules.search
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.Editable
+import android.util.Log
+import android.view.View
+import android.widget.CheckBox
+import butterknife.*
+import com.fastaccess.R
+import com.fastaccess.helper.AnimHelper
+import com.fastaccess.helper.InputHelper
+import com.fastaccess.helper.Logger
+import com.fastaccess.ui.base.BaseActivity
+import com.fastaccess.ui.base.mvp.BaseMvp
+import com.fastaccess.ui.base.mvp.presenter.BasePresenter
+import com.fastaccess.ui.modules.search.repos.SearchReposFragment
+import com.fastaccess.ui.widgets.FontAutoCompleteEditText
+
+class SearchUserActivity : BaseActivity<BaseMvp.FAView, BasePresenter<BaseMvp.FAView>>() {
+
+    companion object {
+
+        @JvmStatic
+        val USERNAME = "username"
+        @JvmStatic
+        val SEARCH_TERM = "search"
+
+        @JvmStatic
+        fun getIntent(context: Context, username: String, searchTerm: String?): Intent {
+            val intent = Intent(context, SearchUserActivity::class.java)
+            intent.putExtra(USERNAME, username)
+            intent.putExtra(SEARCH_TERM, searchTerm)
+            return intent
+        }
+    }
+
+    @BindView(R.id.forkCheckBox) lateinit var forkCheckBox: CheckBox
+    @BindView(R.id.clear) lateinit var clear: View
+    @BindView(R.id.searchEditText) lateinit var searchEditText: FontAutoCompleteEditText
+
+    private var username = ""
+    private var searchTerm = ""
+    private var isFork = true
+    private lateinit var searchReposFragment: SearchReposFragment
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        ButterKnife.bind(this)
+
+        searchReposFragment = SearchReposFragment.newInstance()
+        supportFragmentManager.beginTransaction()
+                .replace(R.id.containerFragment, searchReposFragment, "SearchReposFragment")
+                .commit()
+
+        val args = intent.extras
```

Comment:
try rotate your phone and you will get a crash. 

intent is always null when activity orientation changes. 

use variables initialization and annotated them with `@State` like in any other place and init them only when savedInstance == null

Reply:
I actually tried this. 
The Arguments bundle was saved on my devices.
Tried this on Android O Google Pixel, and Android N Oneplus 5.
Both worked without any crash.

Reply:
That's strange, it should have crashed 🤔. But to be sure and safe please follow what I mentioned.